### PR TITLE
feat: add AllowedMentions.from_message

### LINKED ISSUE #573: MessageInteraction.edit_original_message does not preserve allowed_mentions state if kwarg not passed
### Summary

MessageInteraction.edit_original_message does not preserve allowed_mentions state if kwarg not passed

### Reproduction Steps

1. Set Client.allowed_mentions to AllowedMentions.none()
2. Send a message that mentions a user containing a button, explicitly defining allowed_mentions
3. Click the button, and have the button handler update the content of the message using `inter.edit_original_message(content=...)`
4. Observe that the allowed_mentions is not preserved

### Minimal Reproducible Code

_No response_

### Expected Results

The edited message remains highlighted, mentioning the user

### Actual Results

The edited message no longer appears highlighted

Attaching a debugger reveals that https://github.com/DisnakeDev/disnake/blob/master/disnake/interactions/base.py#L449 became an AllowedMentions.none() instead of the actual previous allowed mentions

### Intents

discord.Intents(     guilds=True,     members=True,     messages=True,     reactions=True,     bans=False,     emojis=False,     integrations=False,     webhooks=False,     invites=False,     voice_states=False,     presences=False,     typing=False, )

### System Information

```markdown
- Python v3.10.2-final
- disnake v2.4.0-final
    - disnake pkg_resources: v2.4.0
- aiohttp v3.8.1
- system info: Darwin 18.7.0 Darwin Kernel Version 18.7.0: Tue Jun 22 19:37:08 PDT 2021; root:xnu-4903.278.70~1/RELEASE_X86_64

Note: this is not the most recent Disnake version but I have confirmed that the applicable code has not been updated since.
```

### Checklist

- [X] I have searched the open issues for duplicates.
- [X] I have shown the entire traceback, if possible.
- [X] I have removed my token from display, if visible.

### Additional Context

- This might only occur if Client.allowed_mentions is set, overwriting the allowed_mentions with Client.allowed_mentions

### PR DESCRIPTION
**Pull request by onerandomusername**
## Summary

superseeds #592 

closes #573

Add `AllowedMentions.from_message()` which creates an AllowedMentions from a message's current state.

## Checklist

- [ ] If code changes were made, then they have been tested
    - [x] I have updated the documentation to reflect the changes
    - [ ] I have formatted the code properly by running `task lint`
    - [ ] I have type-checked the code by running `task pyright`
- [ ] This PR fixes an issue
- [ ] This PR adds something new (e.g. new method or parameters)
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: disnake/mentions.py [hunk 1]
```diff
@@ -111,6 +112,22 @@ def none(cls: Type[A]) -> A:
...
+        Note that this is not what AllowedMentions the message was sent with, but what the message actually mentioned.
+        For example, a message that successfully mentioned everyone will have :attr:`~AllowedMentions.everyone` set to ``True``.
+
+        .. versionadded:: 2.6
+        """
+        return cls(
+            everyone=message.mention_everyone,
+            users=message.mentions.copy(),  # type: ignore # mentions is a list of Snowflakes
+            roles=message.role_mentions.copy(),  # type: ignore # mentions is a list of Snowflakes
+            replied_user=False,
```
Comment (shiftinv) [inline disnake/mentions.py:128]: `replied_user` shouldn't be set to false by default, or alternatively the documentation should mention that caveat. This currently doesn't work, i.e. it no longer mentions the user after editing the message, with `AllowedMentions.none()` set globally:
```py
reply = await ctx.message.reply("h", allowed_mentions=AllowedMentions(replied_user=True))
await asyncio.sleep(1)
await reply.edit("h2", allowed_mentions=AllowedMentions.from_message(reply))
```

I believe the client solves this through setting `replied_user` on edit by checking for `MessageType.reply` and whether the referenced message's author is part of `message.mentions`.

Comment (onerandomusername) [reply disnake/mentions.py:128]: Will update. I thought that providing that mention in the list of user mentions would override this.

Comment (onerandomusername) [reply disnake/mentions.py]: https://github.com/DisnakeDev/disnake/pull/603/commits/b0b2a6d0e34108794e4cb3816373842ace63621a

File: disnake/mentions.py [hunk 2]
```diff
@@ -111,6 +114,30 @@ def none(cls: Type[A]) -> A:
...
+        """
+        # circular import
+        from .message import Message
+
+        return cls(
+            everyone=message.mention_everyone,
+            users=message.mentions.copy(),  # type: ignore # mentions is a list of Snowflakes
+            roles=message.role_mentions.copy(),  # type: ignore # mentions is a list of Snowflakes
+            replied_user=bool(
+                message.type == MessageType.reply
```
Comment (shiftinv) [inline disnake/mentions.py:134]:
```suggestion
                message.type is MessageType.reply
```

**Review by shiftinv — APPROVED** [review]

**Pull request merged**